Makefile

```diff
@@ -140,6 +140,7 @@ base.image.go1.19 := golang:1.19
 base.image.go1.20 := golang:1.20
 base.image.go1.21 := golang:1.21
 base.image.go1.22 := golang:1.22
+base.image.go1.23 := golang:1.23
```

Comment:
I am good to add 1.23 added. 

Meanwhile, according to go maintenance policy `Each major Go release is supported until there are two newer major releases.`. I think we at least only need to release N-4?

Reply:
> go1.19.13 (released 2023-09-06) includes fixes to the go command, and the crypto/tls and net/http packages. See the [Go 1.19.13 milestone](https://github.com/golang/go/issues?q=milestone%3AGo1.19.13+label%3ACherryPickApproved) on our issue tracker for details.

As N-4(1.19) was not released for nearly a year.

Reply:
Mostly, 1.18 should be the lowest version we plan to support.

Reply:
I refer to https://go.dev/doc/devel/release

Reply:
Ok. Then we can remove between the 16-19 versions?

Reply:
According to release note, we could at least remove 16, 17, 18, 19(this could consider to be kept).

Reply:
Sure, I will keep the 19 and higher versions.

Reply:
I will temporarily make this PR a draft to wait for all testing success because all older tests are based on go `1.18`.